src/ruby/ext/grpc/rb_call.c

```diff
@@ -179,6 +179,36 @@ static VALUE grpc_rb_call_cancel(VALUE self) {
   return Qnil;
 }
 
+/* Not a part of the surface API.
+ * This is meant for internal usage by the "write thread" of grpc-ruby
+ * client-side bidi calls. It provides a way for the background write-thread
+ * to propogate failures to the main read-thread and give the user an error
+ * message. */
+static VALUE grpc_rb_call_cancel_with_bidi_write_thread_error_message(
```

Comment:
My hesitation though is just that it seems that the `const char*` type for `description doesn't require that  `grpc_call_cancel_with_status` make a copy of `description` - it seems like the `const char*` type would allow `grpc_call_cancel_with_status` to internally do something like `GRPC_ERROR_CREATE_FROM_COPIED_STRING(description)`.